vortex-ipc/src/reader.rs

```diff
@@ -275,11 +341,20 @@ impl StreamMessageReader {
 #[cfg(test)]
 mod tests {
```

Comment:
Can you add a test with a stream containing two arrays that if you call take(...) on the first one, you can then successfully read the second one. I think right now you leave the first array half-consumed if there are no indices in the final chunk.